compiler/rustc_typeck/src/check/mod.rs

```diff
@@ -520,6 +521,35 @@ fn typeck_with_fallback<'tcx>(
 
             let fn_sig = fixup_opaque_types(tcx, &fn_sig);
 
+            if fn_sig.abi == abi::Abi::RustCall {
+                let expected_args = if let ImplicitSelfKind::None = decl.implicit_self { 1 } else { 2 };
+
+                let err = || {
+                    if let Node::Item(item) = tcx.hir().get(id) {
```

Comment:
https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/hir/map/struct.Map.html#method.expect_item
```suggestion
                    let item = tcx.hir().expect_item(id);
```